rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -1252,6 +1258,193 @@ def __init__(
 
         self._prepare_layers()
 
+    def adjust_sparse_layers(
+        self,
+        data_example: Dict[Text, List[FeatureArray]],
+        new_feature_sizes: Dict[Text, Dict[Text, List[int]]],
+        old_feature_sizes: Dict[Text, Dict[Text, List[int]]],
+    ) -> None:
+        """Adjusts sizes of DenseForSparse layers.
+
+        Updates sizes of DenseForSparse layers by comparing current sparse feature
+        sizes to old ones. This must be done before fine-tuning starts to account for
+        sparse features that might have changed. If any layer is changed, compiles
```

Comment:
```suggestion
        sizes to old ones. This must be done before fine-tuning starts to account for
        any change in the size of sparse features that might have happened because of addition of new data. If any layer is changed, compiles
```